synapse/logging/opentracing.py

```diff
@@ -333,9 +377,160 @@ def inject_active_span_byte_dict(headers, destination):
         headers[key.encode()] = [value.encode()]
 
 
+@only_if_tracing
+def inject_active_span_text_map(carrier, destination=None):
+    if destination and not whitelisted_homeserver(destination):
+        return
+
+    opentracing.tracer.inject(
+        opentracing.tracer.active_span, opentracing.Format.TEXT_MAP, carrier
+    )
+
+
+def active_span_context_as_string():
+    carrier = {}
+    if opentracing:
+        opentracing.tracer.inject(
+            opentracing.tracer.active_span, opentracing.Format.TEXT_MAP, carrier
+        )
+    return json.dumps(carrier)
+
+
+@only_if_tracing
+def span_context_from_string(carrier):
+    carrier = json.loads(carrier)
+    return opentracing.tracer.extract(opentracing.Format.TEXT_MAP, carrier)
+
+
+@only_if_tracing
+def extract_text_map(carrier):
+    return opentracing.tracer.extract(opentracing.Format.TEXT_MAP, carrier)
+
+
+# Tracing decorators
+
+
+def trace_deferred(func):
+    """Decorator to trace a deferred function. Sets the operation name to that of the
+    function's."""
```

Comment:
Would be good to just include a quick example, mainly to make it obvious that some of these decorators take arguments and some don't, e.g.:

```python
@trace_deferred
def some_func(self):
    ...
```

vs

```python
@trace_deferred_using_operation_name("foobar")
def some_func(self):
    ...
```

Reply:
Examples already provided by the docs merged to develop earlier.